cli/azd/internal/repository/app_init.go

```diff
@@ -472,23 +485,74 @@ func (i *Initializer) prjConfigFromDetect(
 					config.Resources["postgres"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbPostgres,
 						Name: spec.DbPostgres.DatabaseName,
+						Props: project.PostgresProps{
+							DatabaseName: spec.DbPostgres.DatabaseName,
+							AuthType: decideAuthType(
+								spec.DbPostgres.AuthUsingManagedIdentity,
+								spec.DbPostgres.AuthUsingUsernamePassword,
+								false),
+						},
 					}
 				case appdetect.DbMySql:
 					config.Resources["mysql"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbMySQL,
 						Props: project.MySQLProps{
 							DatabaseName: spec.DbMySql.DatabaseName,
-							AuthType:     "managedIdentity",
+							AuthType: decideAuthType(
+								spec.DbMySql.AuthUsingManagedIdentity,
+								spec.DbMySql.AuthUsingUsernamePassword,
+								false),
 						},
 					}
 				case appdetect.DbRedis:
 					config.Resources["redis"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbRedis,
 					}
+				case appdetect.DbCosmos:
+					cosmosDBProps := project.CosmosDBProps{
+						DatabaseName: spec.DbCosmos.DatabaseName,
+					}
+					for _, container := range spec.DbCosmos.Containers {
+						cosmosDBProps.Containers = append(cosmosDBProps.Containers, project.CosmosDBContainerProps{
+							ContainerName:     container.ContainerName,
+							PartitionKeyPaths: container.PartitionKeyPaths,
+						})
+					}
+					config.Resources["cosmos"] = &project.ResourceConfig{
+						Type:  project.ResourceTypeDbCosmos,
+						Props: cosmosDBProps,
+					}
+				}
+
+			}
+			for _, azureDep := range prj.AzureDeps {
```

Comment:
Maybe we can merge `prj.DatabaseDeps` and `prj.AzureDeps`. They are all `backing service`s

Reply:
In my understanding, the final simplified code will be like this:
```
type AuthType string

const (
	UserAssignedManagedIdentity AuthType = "User assigned managed identity"
	UsernamePassword            AuthType = "Username and password"
	ConnectionString            AuthType = "Connection String"
)

type AzureDependencyType string

const (
	AzureDependencyTypeAzureEventHubs AzureDependencyType = "Azure Event Hubs"
	// omit Other types like service bus, cosmos db.
)

type AzureDependency interface {
	GetAzureDependencyType() AzureDependencyType
	GetAuthType() AuthType
}

type AzureDependencyAzureEventHubs struct {
	EventHubNames []string
	AuthType      AuthType
	UseKafka      bool
}

func (azureEventHubs *AzureDependencyAzureEventHubs) GetAzureDependencyType() AzureDependencyType {
	return AzureDependencyTypeAzureEventHubs
}

func (azureEventHubs *AzureDependencyAzureEventHubs) GetAuthType() AuthType {
	return azureEventHubs.AuthType
}
```